master/internal/config/config.go

```diff
@@ -236,67 +236,42 @@ type ObservabilityConfig struct {
 	EnablePrometheus bool `json:"enable_prometheus"`
 }
 
-func readPreemptionFromScheduler(conf *resourcemanagers.SchedulerConfig) *bool {
-	if conf == nil || conf.Priority == nil {
-		return nil
-	}
-	return &conf.Priority.Preemption
-}
-
 func readPriorityFromScheduler(conf *resourcemanagers.SchedulerConfig) *int {
 	if conf == nil || conf.Priority == nil {
 		return nil
 	}
 	return conf.Priority.DefaultPriority
 }
 
-// ReadPreemptionStatus resolves the desired preemption status for a job.
-func ReadPreemptionStatus(rpName string, jobConf interface{}) bool {
+// ReadRMPreemptionStatus resolves the preemption status for a resource manager.
+func ReadRMPreemptionStatus(rpName string) bool {
```

Comment:
I think this should be part of the RM interface, whether it supports preemption or not. I don't think callers should be digging into the config to try to figure this out. On top of making it more clear what the RP interface is within the system and encapsulating it all, it's more flexible (e.g. if preemption could be disabled dynamically on RPs or if we don't know if a RP supports preemption or not until we make some API call that would be made by the RM itself)

Reply:
I think this highlights my gripe with the singleton config too; more than likely it is not the right place to get the information; the RM should get its configuration and construct its state, and if you need to know something about that state, you should ask it, not try to infer it from its configuration.